### DISCUSSION TITLE
AWS Backup integrated?

### DISCUSSION THREAD

**Discussion by marijakstrazdas** [gruntwork-io/knowledge-base#64] [Help] [answered]
Does Gruntwork have AWS Backup (the org-wide managed solution) integrated into your process?

**Comment by zackproser** [comment, marked as answer]
We do not currently support AWS Backup. However, all of our data storage modules include configurable snapshot capturing functionality. In addition, we offer special modules for snapshotting AWS Lambda and RDS resources in https://github.com/gruntwork-io/terraform-aws-data-storage.  

That said, because the Reference Architecture is defined entirely within Infrastructure as Code, adding support for AWS Backup using [the AWS Backup Terraform resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/backup_plan) should be straightforward.

**Reply by josh-padnick** [reply]
Do we plan to support AWS Backup as part of our Gruntwork Compliance work?

**Reply by zackproser** [reply]
That's a great question. The honest answer is that I don't **yet** know. We are moving through our repos via vertical slices, starting with the risk analyses that in turn inform the gap analyses that we will ultimately split out into the project plan / tickets. 

As of right now, we've just about finished going through the VPC modules. Once we arrive at the data store repos, I'll be able to tell you if AWS Backup ends being identified as one of the gaps we need to address, as part of the compliance work.

In case we _don't_ end up prioritizing AWS Backup support as part of the compliance work, it would be great to better understand the frequency of user requests for it, so we could better prioritize it against the compliance work.

**Reply by zackproser** [reply]
@josh-padnick @marijakstrazdas I decided to take a stab at [implementing AWS Backup support](https://github.com/gruntwork-io/terraform-aws-data-storage/pull/215/files) in our `terraform-aws-data-storage` repo. 

Once we've gotten it merged there, we'll still need to add support in our service catalog and then integrate it into our deployment tooling, but this should get the ball rolling.
